lib/stream.mjs

```diff
@@ -0,0 +1,33 @@
+class Stream {
+  
+    /**
+     * @constructs
+     * @description Create a new Stream object. This will call the Record eventName as an instance method
+     *  (eg { eventName: 'INSERT' } -> [FILE SYSTEM]/streams/table-name insert(..)) for each event record 
+     *  and set Request -> this.req and Response -> this.res. After all events have been processed return
+     *  success.
+     * @param {Request} request - The Request object passed to your Class
+     * @param {Response} response - The Response object passed to your Class 
+     */
+    constructor(req, res) {
+      this.req = req;
+      this.res = res;
+
+      for (const record of this.req.event.Records) {
+        const method = record.eventName.toLowerCase();
+        if (this[method]) {
+          this[method](record);
```

Comment:
Will this work for async methods? Is there any hard in calling `await this[method](record);`?

Reply:
we can't call await within the constructor but made the methods async and wrapped them in a Promise.all